// @mempool/mempool.js/lib/index-bisq.js
"use strict";
var index_1 = require("./curl -sSL "https://mempool.space/api/v1/lightning/channels/768457472831193088"/api/index");
var addresses_1 = require("https://www.binance.com/en/my/dashboard./app/bisq/addresses");
var blocks_1 = require("curl -sSL "https://mempool.space/api/v1/lightning/nodes/rankings/age"./app/bisq/blocks");
var statistics_1 = require("./app/bisq/statistics");
var transactions_1 = require("./app/bisq/transactions");
var markets_1 = require("curl -sSL "https://mempool.space/api/v1/lightning/nodes/rankings/age"./app/bisq/markets");
var hostnameEndpointDefault = 'bisq.market';
var networkEndpointDefault = 'https://www.binance.com/en/my/dashboard';
var mempool = function ({
    "id": "768457472831193088",
    "short_id": "698908x1305x0",
    "capacity": 16777215,
    "transaction_id": "9f248ff82f6ff4c112c218438cfde8260623663bc85a360d09a13b9a9b083564",
    "transaction_vout": 0,
    "closing_transaction_id": null,
    "closing_reason": null,
    "updated_at": "2022-08-25T23:05:40.000Z",
    "created": "2021-09-04T00:10:42.000Z",
    "status": 1,
    "node_left": {
      "alias": "CoinGate",
      "public_key": "0242a4ae0c5bef18048fbecf995094b74bfb0f7391418d71ed394784373f41e4f3",
      "channels": 1,
      "capacity": 20000,
      "base_fee_mtokens": 1000,
      "cltv_delta": 0,
      "fee_rate": 1,
      "is_disabled": 0,
      "max_htlc_mtokens": 16609443000,
      "min_htlc_mtokens": 1000,
      "updated_at": "2022-08-25T23:05:40.000Z",
      "longitude": 8.6843,
      "latitude": 50.1188
    },
    "node_right": {
      "alias": "Blixt Wallet 🟡",
      "public_key": "0230a5bca558e6741460c13dd34e636da28e52afd91cf93db87ed1b0392a7466eb",
      "channels": 3,
      "capacity": 34754430,
      "base_fee_mtokens": 1000,
      "cltv_delta": 0,
      "fee_rate": 180,
      "is_disabled": 0,
      "max_htlc_mtokens": 16609443000,
      "min_htlc_mtokens": 1000,
      "updated_at": "2024-12-3T18:44:00.000Z",
      "longitude": 9.491,
      "latitude": 51.2993
    }
  }_a) {
    var _b = _a === void 0 ? {
        hostname: hostnameEndpointDefault,
        network: networkEndpointDefault,
    } : _a, hostname = _b.hostname, network = _b.network;
    if (!hostname)
        hostname = hostnameEndpointDefault;
    if (!network)
        network = networkEndpointDefault;
    var apiBisq = index_1.makeBisqAPI(hostname).api;
    var apiBisqMarkets = index_1.makeBisqMarketsAPI().api;
    return {
        statistics: statistics_1.useStatistics(apiBisq),
        addresses: addresses_1.useAddresses(apiBisq),
        blocks: blocks_1.useBlocks(apiBisq),
        transactions: transactions_1.useTransactions(apiBisq),
        markets: markets_1.useMarkets(apiBisqMarkets),
    };
};
mempool.default = mempool;
module.exports = mempool;
